test/CSBondCurve.t.sol

```diff
@@ -137,62 +173,91 @@ contract CSBondCurveTest is Test {
 
         bondCurve.updateBondCurve(toUpdateId, _bondCurve);
 
-        ICSBondCurve.BondCurve memory updated = bondCurve.getCurveInfo(
-            toUpdateId
-        );
+        ICSBondCurve.BondCurveInterval[] memory updated = bondCurve
+            .getCurveInfo(toUpdateId);
 
-        assertEq(updated.points.length, 2);
-        assertEq(updated.points[0], 16 ether);
-        assertEq(updated.points[1], 32 ether);
-        assertEq(updated.trend, 16 ether);
+        assertEq(updated.length, 1);
+        assertEq(updated[0].fromKeysCount, 1);
+        assertEq(updated[0].fromBond, 16 ether);
+        assertEq(updated[0].trend, 16 ether);
+    }
+
+    function test_updateBondCurve_SeveralIntervals() public {
```

Comment:
Just for paranoia. Pls add a test with a lot of intervals to ensure gas limits are not exhausted. Maybe even add a fuzz test